### PR TITLE
Quasar 17.8.0 plat 23082

### PR DESCRIPTION
**Pull request by danbar0kal**

### CODE REVIEW

File: release-notes.md [hunk 1]
```diff
@@ -1,3 +1,11 @@
+# Quasar-17.9.0 #
+## Partner registration - Move all time consuming jobs to run in batch ##
+Cloning of UICONF and Category takes 4 seconds, moving them to batch
```
Comment (inbal-ben-david) [inline release-notes.md:3]: write a more general description, the fact that it takes 4 sec is relevant to a specific copy case

File: batch/batches/CopyPartner/KAsyncCopyPartner.class.php [hunk 2]
```diff
@@ -86,5 +107,122 @@ protected function copyAllEntries()
...
+		$newCategory = clone($category);
+		if ($newCategory->parentId != 0)
+		{
+			$newCategory->parentId = $parentCategoryIdMapping[$category->parentId];
+		}
+		$newCategory->id = null;
+		$newCategory->depth = null;
+		$newCategory->partnerId = null;
+		$newCategory->fullName = null;
+		$newCategory->fullIds = null;
```
[the remark this answers is not in the document]
Comment (danbar0kal) [reply batch/batches/CopyPartner/KAsyncCopyPartner.class.php:162]: a clone action is not available via the api, so i am creating a totally new category with the essential data and use the add action. the hierarchy is created according to the original category

File: release-notes.md (hunk 1 above)
Comment (danbar0kal) [reply release-notes.md:3]: done

File: batch/batches/CopyPartner/KAsyncCopyPartner.class.php [hunk 3]
```diff
@@ -86,5 +107,122 @@ protected function copyAllEntries()
...
+			$receivedObjectsCount = $uiConfList->objects ? count($uiConfList->objects) : 0;
+			$pageFilter->pageIndex++;
+			
+			if ( $receivedObjectsCount > 0 )
+			{
+				// Write the source partner's entries to the destination partner
+				self::impersonate( $this->toPartnerId );
+				foreach ( $uiConfList->objects as $uiConf )
+				{
+					$result = $this->getClient()->uiConf->add( $this->cloneUiConf($uiConf));
```
[the remark this answers is not in the document]
Comment (danbar0kal) [reply batch/batches/CopyPartner/KAsyncCopyPartner.class.php:208]: i need to switch between partner clients during the loop. list fromPartner add toPartner

File: batch/batches/CopyPartner/KAsyncCopyPartner.class.php [hunk 4]
```diff
@@ -86,5 +107,122 @@ protected function copyAllEntries()
...
+		
+		$pageFilter = new KalturaFilterPager();
+		$pageFilter->pageSize = 50;
+		$pageFilter->pageIndex = 1;
+		/* @var $this->getClient() KalturaClient */
+		do
+		{
+			// Get the source partner's entries list
+			self::impersonate($this->fromPartnerId);
+			$client = $this->getClient()->uiConf;
```
[the remark this answers is not in the document]
Comment (danbar0kal) [reply batch/batches/CopyPartner/KAsyncCopyPartner.class.php:196]: cant. i need to switch between partner clients during the loop. list fromPartner add toPartner

File: batch/batches/CopyPartner/KAsyncCopyPartner.class.php [hunk 5]
```diff
@@ -86,5 +107,122 @@ protected function copyAllEntries()
...
+	}
+	
+	protected function cloneCategory($category, &$parentCategoryIdMapping)
+	{
+		/* @var $category KalturaCategory */
+		$newCategory = clone($category);
+		if ($newCategory->parentId != 0)
+		{
+			$newCategory->parentId = $parentCategoryIdMapping[$category->parentId];
+		}
```
[the remark this answers is not in the document]
Comment (danbar0kal) [reply batch/batches/CopyPartner/KAsyncCopyPartner.class.php:157]: they are being set when the new category is created

File: batch/batches/CopyPartner/KAsyncCopyPartner.class.php [hunk 6]
```diff
@@ -86,5 +107,122 @@ protected function copyAllEntries()
...
 		self::unimpersonate();
-	}	
+	}
+	
+	protected function copyCategories()
+	{
+		$this->log("Copying categories from partner [" . $this->fromPartnerId . "] to partner [" . $this->toPartnerId . "]");
+		
+		$categoryFilter = new KalturaCategoryFilter();
+		$categoryFilter->order = self::ID;
```
[the remark this answers is not in the document]
Comment (danbar0kal) [reply batch/batches/CopyPartner/KAsyncCopyPartner.class.php:117]: no reason. i removed it.

**Comment by inbal-ben-david** [comment]
reviewed

**Pull request merged**